server/lib/applications.coffee

```diff
@@ -0,0 +1,202 @@
+                    slug: app.slug
+                    password: app.password
+
+                # Extract application metadata from manifest
+                data.widget = manifest.getWidget()
+                data.version = manifest.getVersion()
+                data.iconPath = manifest.getIconPath()
+                data.color = manifest.getColor()
+                data.needsUpdate = false
+                infos =
+                    git: app.git
+                    name: app.name
+                    icon: app.icon
+                    iconPath: data.iconPath
+                    slug: app.slug
+
+                # Set icons information.
+                try
+                    iconInfos = icons.getIconInfos infos
+                catch err
+                    console.log err if process.env.NODE_ENV isnt 'test'
+                    iconInfos = null
+                data.iconType = iconInfos?.extension or null
+
+                # Run the application process based on collected data.
+                appHelpers._runUpdate app, data, access, callback
+
+
+    # Request controller for installation.
+    _runInstall: (appli, callback) ->
+        manager.installApp appli, (err, result) ->
+            if err
+                appHelpers.markBroken appli, err
+                console.log err if err
+
+            else if not result.drone?
+                err = new Error(
+                    "Controller didn't return informations about #{appli.name}."
+                )
+                console.log err if err
+
+            else
+                if result.drone.type is 'static'
+                    appli.type = result.drone.type
+                    appli.path = result.drone.path
+                    msg = "Static app successfully installed."
+                else
+                    appli.port = result.drone.port
+                    msg = "Install succeeded, app is running on port "
+                    msg += "#{appli.port}."
+
+                log.info msg
+                callback appli
+
+
+    # Run Update process based on given metadata:
+    #
+    # * Request the controller to run the update.
+    # * Persist metadata.
+    _runUpdate: (app, data, access, callback) ->
+
+        # Update access token.
+        app.updateAccess access, (err) ->
+            return callback err if err?
+
+            # Update application
+            manager.updateApp app, (err, result) ->
+                return callback err if err?
+                data.state = "installed" if app.state isnt "stopped"
+
+                # Update metadata and icon.
+                app.updateAttributes data, (err) ->
+                    removeAppUpdateNotification app
+
+                    icons.save app, iconInfos, (err) ->
+                        if err and process.env.NODE_ENV isnt 'test'
+                            console.log err.stack
+                        else
+                            console.info 'icon attached'
```

Comment:
Maybe use `log.info` instead of `console.log` here. And remove the condition on NODE_ENV.